syntax = "proto2";

package PiNetworkCommand;

enum PiNetworkStatus {
    OK = 1;
    ERROR = 2;
    ERROR_UNKNOWN_INTERFACE = 3;
}

message GetIPAddress {
    required string interface = 1;

    message Response {
        required PiNetworkStatus status = 1;
        required string address = 2;
    }
}

message EnableInterface {
    required string interface = 1;
    required bool value = 2;

    message Response {
        required PiNetworkStatus status = 1;
    }
}

message PiNetworkCommand {
    oneof command {
        GetIPAddress getIPAddress = 1;
        EnableInterface enableInterface = 2;
    }
}
